### PR TITLE
Update codegen image version

### PR DESCRIPTION
**Pull request by ffjlabo**
**What this PR does / why we need it**:

Update codegen image version on `gen/code`

**Which issue(s) this PR fixes**:

Part of https://github.com/pipe-cd/pipecd/issues/4874
Related to https://github.com/pipe-cd/pipecd/pull/4968

**Does this PR introduce a user-facing change?**:

- **How are users affected by this change**:
- **Is this breaking change**:
- **How to migrate (if breaking change)**:

### CODE REVIEW

**Comment by ffjlabo** [comment]
gen/code failed 😇 I will investigate it.
https://github.com/pipe-cd/pipecd/actions/runs/9492831530/job/26160595802?pr=4970

File: Makefile [hunk 1]
```diff
@@ -199,7 +199,7 @@ update/copyright:
 .PHONY: gen/code
 gen/code:
 	# NOTE: Specify a specific version temporally until the next release.
-	docker run --rm -v ${PWD}:/repo -it --entrypoint ./tool/codegen/codegen.sh ghcr.io/pipe-cd/codegen@sha256:fc3db505ef8dbf287b90aafed8c28246d2cca06bda2b43893a3059719fe9fff4 /repo #v0.44.0-38-g7229285
+	docker run --rm -v ${PWD}:/repo -it --entrypoint ./tool/codegen/codegen.sh ghcr.io/pipe-cd/codegen:v0.47.3-rc0-1-g7e27cad /repo
```
Comment (Warashi) [inline Makefile:202]: How about specifying the image with sha256 hash, as specified in the PR's base commit?

Comment (ffjlabo) [reply Makefile:202]: I see.
Is the sha256 hash better than specifying version?
I don't have a strong opinion, but I felt redundant hash with version comment like the base commit.

**Comment by khanhtc1202** [comment]
@ffjlabo You have to update this line in order to make the CI passed
https://github.com/pipe-cd/pipecd/blob/master/.github/workflows/gen.yaml#L14

**Comment by ffjlabo** [comment]
oops sorry 🙏

File: Makefile (hunk 1 above)
Comment (Warashi) [reply Makefile:202]: The image specified with the tag can be overwritten, such as the git tag.
Conversely, the sha256 hash is immutable, such as the git commit hash.
So, specifying with the sha256 hash is more reproducible.

Comment (ffjlabo) [reply Makefile:202]: I got your point! Thank you! I'll fix it.

**Comment by ffjlabo** [comment]
Workflow still failed 🤔 
https://github.com/pipe-cd/pipecd/actions/runs/9492993824
```
docker://ghcr.io/pipe-cd/codegen@sha256:277a911717e5bd33aca7b38f05dd8ab4a047ba5038a81f9b43d816a5a921fe84 is not allowed to be used in pipe-cd/pipecd. Actions in this workflow must be: within a repository that belongs to your Enterprise account or matching the following: actions/cache@v2, actions/setup-go@v2, Azure/setup-helm@v1, actions/setup-node@v3, peaceiris/actions-hugo@c03b5dbed22245418539b65eb9a3b1d5fdd9a0a6, softprops/action-gh-release@1e07f4398721186383de40550babbdf2b84acfc5, actions/labeler@v4, reviewdog/action-golangci-lint@f3dc5fadcaff5d8da3574b129a58db433171b1a8, actions/checkout@v3, docker/login-action@49ed152c8eca782a232dede0303416e8f356c37b, actions/setup-go@v3, contributor-assistant/github-action@b3bbab0a75fa27270069933cec6f369c0b373b4e, pipe-cd/actions-gh-release@v2.4.0, github/codeql-action/analyze@v2, github/codeql-action/autobuild@v2, github/codeql-action/init@v2, codecov/codecov-action@v3, docker/setup-qemu-action@8b122486cedac8393e77aa9734c3528886e4a1a8, docker...
```

> docker://ghcr.io/pipe-cd/codegen@sha256:277a911717e5bd33aca7b38f05dd8ab4a047ba5038a81f9b43d816a5a921fe84 is not allowed to be used in pipe-cd/pipecd. Actions in this workflow must be: within a repository that belongs to your Enterprise account or matching the following

Maybe need to fix the actions setting for the repo

**Comment by ffjlabo** [comment]
I tried to add the `docker://ghcr.io/pipe-cd/codegen@sha256:277a911717e5bd33aca7b38f05dd8ab4a047ba5038a81f9b43d816a5a921fe84` to the `Actions permissions` in the setting.

So lets recheck.

**Comment by ffjlabo** [comment]
CI failed because there is the diff in `pkg/app/server/service/webservice/service.pb.auth.go`.

https://github.com/pipe-cd/pipecd/actions/runs/9493373712/job/26161960073?pr=4970
```
HEAD detached at pull/4970/merge
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   pkg/app/server/service/webservice/service.pb.auth.go
```

**Review by khanhtc1202 — DISMISSED** [review]

**Comment by khanhtc1202** [comment]
So with the new codegen image, the generated auth file still not the same as last time succeed 🤔 Which diff is generated? 👀

**Comment by ffjlabo** [comment]
@khanhtc1202 
This is the diff. Some methods disappear 🤔 
less

**Comment by khanhtc1202** [comment]
So this issue remained 🥹
https://github.com/pipe-cd/pipecd/issues/4963

**Comment by ffjlabo** [comment]
CI `gen` passed!
https://github.com/pipe-cd/pipecd/actions/runs/9509753883/job/26213180311?pr=4970

File: .github/workflows/gen.yaml [hunk 2]
```diff
@@ -10,8 +10,11 @@ jobs:
     steps:
     - name: Checkout repository
       uses: actions/checkout@v3
+    - name: check CPU architecture
+      shell: bash
+      run: cat /proc/cpuinfo && uname -a
```
Comment (khanhtc1202) [inline .github/workflows/gen.yaml:15]: Do we still need this?

Comment (ffjlabo) [reply .github/workflows/gen.yaml]: Sorry, I forgot to remove this.
removed [b06e221](https://github.com/pipe-cd/pipecd/pull/4970/commits/b06e22198f322bf8f866fdc045ad0094a6bc7f0f)

Comment (Warashi) [inline .github/workflows/gen.yaml]: This step is for debugging?
We can add this on demand, so we don't have to include this step in the usual run.

Comment (Warashi) [reply .github/workflows/gen.yaml]: oops, you already removed this. thank you.

**Review by Warashi — APPROVED** [review]
Thank you!

**Review by khanhtc1202 — APPROVED** [review]
Here you go 👍

**Pull request merged**